### DISCUSSION TITLE
25 result items hard limit

### DISCUSSION THREAD

**Discussion by hedgieinsocks** [Ulauncher/Ulauncher#1402] [General]
Ulauncher shows scrollable results, and since the default workflow is bound to jump keys, it makes sense to [limit](https://github.com/Ulauncher/Ulauncher/blob/98d07cebdd4868302cf646ce1b41562d4aecaa1c/ulauncher/ui/windows/ulauncher_window.py#L336) the results to the number of jump keys. But for those who disable jump keys, ulauncher hardcodes it to only 25 items. Is there any particular reason for such a number? I recognize that it's kinda logical to type more characters to get to the target item faster instead of scrolling. But perhaps we could tweak it in the settings if there will be no performance penalties wdyt? I'm perfectly good with 25 but posting here just in case this detail is simply overlooked from the legacy code.

**Comment by friday** [comment]
It should affect performance, because we use generators to yield the results, and limiting means less operations that have to be done, and less UI elements that has to be rendered also.

But I never measured it, so it's all premature optimizations. The difference between 25 and 50 is probably negligible. In case you have a huge screen that fits more than 25 results (my screen is 2880x1800 at 200% scale and I can only fit in 13 results) I think raising the upper limit makes sense. However, if you want to remove the upper limit completely and be able to scroll through the results, you are definitely using Ulauncher in a way it wasn't intended, and I think that could noticeably affect performance.
